src/sage/schemes/riemann_surfaces/riemann_surface.py

```diff
@@ -3990,6 +4013,806 @@ def vectomat(v):
+                    A[m + j, c] -= q
+                    K = symplectic_matrix("3", A, g)
+                    M_curr *= K
+                    T *= K
+
+        # Make the pivot positive
+        if M_curr[r, m + j] == -1:
+            A = Matrix.identity(ZZ, g)
+            A[m + j, m + j] = -1
+            K = symplectic_matrix("3", A, g)
+            M_curr *= K
+            T *= K
+
+        # Now we have pivot, we clear
+        p_left = m + j
+        s_left = int(M_curr[r, p_left])
+
+        # pivot in left tail
+        if abs(s_left) == 1:
+            # (i) clear entries to the right inside the left tail
+            for k in range(m + j + 1, g):
+                val = int(M_curr[r, k])
+                if val == 0:
+                    continue
+                A = Matrix.identity(ZZ, g)
+                A[p_left, k] -= val
+                K = symplectic_matrix("3", A, g)
+                M_curr *= K
+                T *= K
+
+            # clear entire right tail
+            for t in range(g + m, 2 * g):
+                val = int(M_curr[r, t])
+                if val == 0:
+                    continue
+                tau = t - g
+                Sg = Matrix.zero(ZZ, g, g)
+                sign = 1 if s_left == 1 else -1
+                Sg[p_left, tau] = -val * sign
+                Sg[tau, p_left] = -val * sign
+                K = symplectic_matrix("1", Sg, g)
+                M_curr *= K
+                T *= K
+
+            # clear g+m+j
+            for s in range(m, 2 * m):
+                val = int(M_curr[s, g + m + j])
+                if val == 0:
+                    continue
+
+                d_s = int(delta[s - m, s - m])
+                if val % d_s != 0:
+                    raise ValueError(
+                        f"Divisibility failed at row {s}: {val} not divisible by {d_s}."
+                    )
+
+                q = val // d_s
+                U = Matrix.identity(ZZ, g)
+                U[m + j, s - m] += q
+                K = symplectic_matrix("3", U, g)
+                M_curr *= K
+                T *= K
+
+            # clear left half of tail inside the left tail
+            for k in range(m, m + j):
+                val = int(M_curr[r, k])
+                if val == 0:
+                    continue
+                if k == p_left:
+                    continue
+                A = Matrix.identity(ZZ, g)
+                A[p_left, k] -= val
+                K = symplectic_matrix("3", A, g)
+                M_curr *= K
+                T *= K
+
+    # Step 5: Clear the lower left m * m block Y
+
+    # Y is unimodular?
+    Y = M_curr[m : 2 * m, 0:m]
+    if not Y.is_zero():
+        # Create the clearing matrix as a block matrix
+        E = Matrix.identity(ZZ, m)
+        zero = Matrix.zero(ZZ, m, m)
+        clear_matrix = block_matrix([[E, zero], [-Y, E]])
+
+        # Apply
+        M_curr = clear_matrix * M_curr
+        S = clear_matrix * S
+
+    N = M_curr
```

Comment:
There is no need for this line as you just retun `N` in the next two lines. If you wanted to help the readability of your code, just make this a comment that `M_curr` is playing the role of N.